### DISCUSSION TITLE
说明文档中需要补充系统环境变量的设置方法，最好补充命令语句

### DISCUSSION THREAD

**Discussion by lucas-wu-wakakak** [xmake-io/xmake#2987] [Q&A]
**原手册：**

我们可以执行下面的命令，获取所有 xmake 用到的环境变量，以及当前被设置的值。

$ xmake show -l envs
XMAKE_RAMDIR            Set the ramdisk directory.
                        <empty>
XMAKE_GLOBALDIR         Set the global config directory of xmake.
                        /Users/ruki/.xmake
XMAKE_ROOT              Allow xmake to run under root.
                        <empty>
XMAKE_COLORTERM         Set the color terminal environment.
                        <empty>
XMAKE_PKG_INSTALLDIR    Set the install directory of packages.
                        <empty>
XMAKE_TMPDIR            Set the temporary directory.
                        /var/folders/vn/ppcrrcm911v8b4510klg9xw80000gn/T/.xmake501/211104
XMAKE_PKG_CACHEDIR      Set the cache directory of packages.
                        <empty>
XMAKE_PROGRAM_DIR       Set the program scripts directory of xmake.
                        /Users/ruki/.local/share/xmake
XMAKE_PROFILE           Start profiler, e.g. perf, trace.
                        <empty>
XMAKE_RCFILES           Set the runtime configuration files.

XMAKE_CONFIGDIR         Set the local config directory of project.
                        /Users/ruki/projects/personal/xmake-docs/.xmake/macosx/x86_64
XMAKE_LOGFILE           Set the log output file path.
                        <empty>
Copy to clipboardErrorCopied
[XMAKE_RAMDIR](https://xmake.io/#/zh-cn/guide/configuration?id=xmake_ramdir)
设置 ramdisk 目录路径
ramdisk 目录是内存文件系统的目录位置，通常 os.tmpdir() 接口会用到，xmake 内部使用的临时文件，如果用户设置 ramdisk 路径，则会优先存储在这个上面，提升整体编译速度。

[XMAKE_TMPDIR](https://xmake.io/#/zh-cn/guide/configuration?id=xmake_tmpdir)
设置用户的临时目录
默认 xmake 会使用 /tmp/.xmake 和 %TEMP%/.xmake，当然用户可以通过这个变量去修改默认路径。

[XMAKE_CONFIGDIR](https://xmake.io/#/zh-cn/guide/configuration?id=xmake_configdir)
设置本地工程配置目录
每个项目的本地编译配置，默认会存储在当前项目根目录的 .xmake 路径下，然后根据不同的平台，架构区分，例如：

.xmake/macosx/x86_64
Copy to clipboardErrorCopied
我们如果不想存储在项目根目录，也可以自己设置到其他路径，比如 build 目录下等等。

**问题：**

**这个里面都没有提到具体这些命令怎么生效的，是直接输入命令配置还是在脚本中用语句配置，麻烦补充下。****

**Comment by waruqi** [comment]
https://xmake.io/#/zh-cn/guide/configuration?id=%e7%8e%af%e5%a2%83%e5%8f%98%e9%87%8f

**Comment by xq114** [comment]
环境变量是专有名词，百度一下就有的，也没有别的含义，这个也解释的话要解释的可太多了
